map/bookmark_manager.cpp

```diff
@@ -1841,15 +1841,15 @@ void BookmarkManager::LoadBookmarkRoutine(std::string const & filePath, bool isT
     auto collection = std::make_shared<KMLDataCollection>();
 
     // Convert KML/KMZ/KMB files to temp KML file and GPX to temp GPX file.
-    std::string fileSavePath = GetKMLorGPXPath(filePath);
-    if (!fileSavePath.empty())
+    std::vector<std::string> pathsToLoad = GetPathsToLoad(filePath);
+    for (auto pathToLoad : pathsToLoad)
```

Comment:
No need to allocate the memory and create a copy of a string.
```suggestion
    for (auto const & pathToLoad : pathsToLoad)
```

Reply:
Does fileToLoad match the context better?

Reply:
Fixed